src/device-state.coffee

```diff
@@ -143,11 +144,94 @@ module.exports = class DeviceState extends EventEmitter
 			applyTargetHealthy = conf.offlineMode or !@applyInProgress or @applications.fetchesInProgress > 0 or cycleTimeWithinInterval
 			return applyTargetHealthy
 
-	normaliseLegacy: =>
+	migrateLegacyApps: (balenaApi) =>
+		console.log('Migrating ids for legacy app...')
+		@db.models('app').select()
+		.then (apps) =>
+			if apps.length == 0
+				console.log('No app to migrate')
+				return
+			app = apps[0]
+			services = JSON.parse(app.services)
+			# Check there's a main service, with legacy-container set
+			if services.length != 1
+				console.log("App doesn't have a single service, ignoring")
+				return
+			service = services[0]
+			if !service.labels['io.resin.legacy-container'] and !service.labels['io.balena.legacy-container']
+				console.log('Service is not marked as legacy, ignoring')
+				return
+			console.log("Getting release #{app.commit} for app #{app.appId} from API")
+			balenaApi.get(
+				resource: 'release'
+				options:
+					filter:
+						belongs_to__application: app.appId
+						commit: app.commit
+						status: 'success'
+					expand:
+						contains__image: [ 'image' ]
+			)
+			.then (releasesFromAPI) =>
+				if releasesFromAPI.length == 0
+					throw new Error('No compatible releases found in API')
+				release = releasesFromAPI[0]
+				releaseId = release.id
+				image = release.contains__image[0].image[0]
```

Comment:
We should check that images are returned from the api here, there always should be, but it's worth checking.
e.g.
```
image = _.get(relase, 'contains__image[0].image[0]')
throw new Error(...) if not image?